packages/injective-math/src/fp_decimal/exp.rs

```diff
@@ -79,293 +99,333 @@ impl FPDecimal {
+                if exponent == FPDecimal::from(51u128) {
                     return Ok(FPDecimal::from_str("1000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(52u128) {
+                if exponent == FPDecimal::from(52u128) {
                     return Ok(FPDecimal::from_str("10000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(53u128) {
+                if exponent == FPDecimal::from(53u128) {
                     return Ok(FPDecimal::from_str("100000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(54u128) {
+                if exponent == FPDecimal::from(54u128) {
                     return Ok(FPDecimal::from_str("1000000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(55u128) {
+                if exponent == FPDecimal::from(55u128) {
                     return Ok(FPDecimal::from_str("10000000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(56u128) {
+                if exponent == FPDecimal::from(56u128) {
                     return Ok(FPDecimal::from_str("100000000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(57u128) {
+                if exponent == FPDecimal::from(57u128) {
                     return Ok(FPDecimal::from_str("1000000000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(58u128) {
+                if exponent == FPDecimal::from(58u128) {
                     return Ok(FPDecimal::from_str("10000000000000000000000000000000000000000000000000000000000").unwrap());
                 }
-                if rhs == FPDecimal::from(59u128) {
+                if exponent == FPDecimal::from(59u128) {
                     return Ok(FPDecimal::from_str("100000000000000000000000000000000000000000000000000000000000").unwrap());
                 }
             }
 
-            fn inner(mut x: FPDecimal, mut n: FPDecimal) -> Result<FPDecimal, OverflowError> {
-                if n == FPDecimal::zero() {
-                    return Ok(FPDecimal::one());
-                }
-
-                let mut y = FPDecimal::one();
-
-                while n > FPDecimal::one() {
-                    if n.num % FPDecimal::TWO.num == FPDecimal::zero().num {
-                        x = x * x;
-                        n /= FPDecimal::TWO;
-                    } else {
-                        y = x * y;
-                        x = x * x;
-                        n = (n - FPDecimal::ONE) / FPDecimal::TWO;
+            fn inner(mut a: FPDecimal, mut b: FPDecimal) -> Result<FPDecimal, OverflowError> {
```

Comment:
I would love an explanation exactly on:

1. what was wrong previously
2. how the fixes solve it

And lastly this code here could be cleaned up a bit.